datastore/postgres/registerscanners.go

```diff
@@ -82,5 +83,60 @@ SELECT
 		registerScannerDuration.WithLabelValues("insert").Observe(time.Since(start).Seconds())
 	}
 
+	return s.populateScanners(ctx)
+}
+
+const selectAllScanner = `
+SELECT
+	id, name, version, kind
+FROM
+	scanner;
+`
+
+func (s *IndexerStore) populateScanners(ctx context.Context) error {
+	s.scanners = make(map[string]int64)
+	rows, err := s.pool.Query(ctx, selectAllScanner)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve scanners: %w", err)
+	}
+	for rows.Next() {
```

Comment:
I think we still need a `defer rows.Close()` (I don't think it'll close if `rows.Scan` fails) and a `rows.Err()` check, too